modules/47_regipol/regiCarbonPrice/datainput.gms

```diff
@@ -22,32 +22,35 @@ $IFTHEN.emiMkt not "%cm_emiMktTarget%" == "off"
 *** initialize emiMkt Target parameters
   p47_targetConverged(ttot2,ext_regi) = 0;
 
-*** initialize carbon taxes before start year 
-if ( (cm_startyear gt 2005),
-  Execute_Loadpoint 'input_ref' p47_taxCO2eq_ref = pm_taxCO2eq;
-  Execute_Loadpoint 'input_ref' p47_taxemiMkt_init = pm_taxemiMkt;
-
-*** copying taxCO2eq value to emiMkt tax parameter for fixed years that contain no pm_taxemiMkt value
-  p47_taxemiMkt_init(ttot,regi,emiMkt)$((p47_taxCO2eq_ref(ttot,regi)) and (ttot.val le cm_startyear) and (NOT(p47_taxemiMkt_init(ttot,regi,emiMkt)))) = p47_taxCO2eq_ref(ttot,regi);
-
-*** Initializing European ETS historical and reference prices
-  loop(regi$regi_groupExt("EUR_regi",regi),
-    p47_taxemiMkt_init("2005",regi,"ETS")$(cm_startyear le 2005) = 0;
-    p47_taxemiMkt_init("2010",regi,"ETS")$(cm_startyear le 2010)  = 15*sm_DptCO2_2_TDpGtC;
-    p47_taxemiMkt_init("2015",regi,"ETS")$(cm_startyear le 2015)  = 8*sm_DptCO2_2_TDpGtC;
-***  p47_taxemiMkt_init("2020",regi,"ETS")$(cm_startyear le 2020)  = 41.28*sm_DptCO2_2_TDpGtC; !! 2018 =~ 16.5€/tCO2, 2019 =~ 25€/tCO2, 2020 =~ 25€/tCO2, 2021 =~ 53.65€/tCO2, 2022 =~ 80€/tCO2 -> average 2020 = 40€/tCO2 -> 40*1.032 $/tCO2 = 41.28 $/t CO2
-    p47_taxemiMkt_init("2020",regi,"ETS")$(cm_startyear le 2020)  = 30*sm_DptCO2_2_TDpGtC;
-
-*** Initializing European ESR historical and reference prices
-    p47_taxemiMkt_init("2020",regi,"ES")$(cm_startyear le 2020)  = 30*sm_DptCO2_2_TDpGtC;
-    p47_taxemiMkt_init("2020",regi,"other")$(cm_startyear le 2020)  = 30*sm_DptCO2_2_TDpGtC;
-  );
+*** initialize carbon taxes
+p47_taxemiMkt_init(t,regi,emiMkt) = 0;
+
+Execute_Loadpoint 'input_ref' p47_taxCO2eq_ref = pm_taxCO2eq;
+Execute_Loadpoint 'input_ref' p47_taxemiMkt_init = pm_taxemiMkt;
+
+*** copying taxCO2eq value to emiMkt tax parameter for years and regions that contain no pm_taxemiMkt value
```

Comment:
I don't fully understand this - what is the difference between the pm_taxCO2eq and pm_taxemiMkt that are read in from the reference gdx? why is p47_taxemiMkt_init filled with pm_taxCO2eq if there is no value in pm_taxemiMkt? 

is the idea to be able to start runs with ETS/ESR split even from a run that didn't have this market split, or am I misunderstanding this? I would find some more detailed comments on this helpful.

Reply:
- `pm_taxCO2eq` is the REMIND default CO2 tax parameter used in the carbon price module. It does not include emission market dimensions (ETS, ESR and other).
- `pm_taxmiMkt` is the CO2 tax parameter used for regional targets in the regipol module. It does include emission market dimensions (ETS, ESR and other).

Only one of them will be active at any time in the code for each region.
`pm_taxCO2eq` is a special case of `pm_taxemiMkt` with just one less dimension. So, all REMIND code could be adapted to use `pm_taxemiMkt` only, but this was never done for practical reasons (and lack of time).

Whenever you define that a region will have its carbon price determined by the regipol module, all information from previous runs need to be moved to the `pm_taxemiMkt` parameter for this specific region, this is why the `p47_taxemiMkt_init` parameter exists. This parameter copies any CO2 tax information from previous runs executed either using regipol or carbonprice modules to be used in the initialization of the scenario carbon taxes.